astrolabe/utils.py

```diff
@@ -266,7 +268,41 @@ def terminate(self):
         # terminated earlier than they actually terminate on Windows.
         if self.is_windows:
             sleep(2)
+            
+        return self.read_stats()
+        
+    def terminate(self):
+        '''Ensure the process is terminated.'''
+        
+        LOGGER.info("Stopping workload executor [PID: {}]".format(self.pid))
+
+        try:
```

Comment:
This try...except seems to be the only difference between the functions... can you explain what we can't just have one function for this?

Reply:
Both functions combined into one per our discussion.

Reply:
👍🏽